packages/babel-helper-create-class-features-plugin/src/fields.js

```diff
@@ -300,8 +303,47 @@ function buildPrivateInstanceFieldInitSpec(ref, prop, privateNamesMap) {
 }
 
 function buildPrivateStaticFieldInitSpec(prop, privateNamesMap) {
-  const { id } = privateNamesMap.get(prop.node.key.id.name);
+  const privateName = privateNamesMap.get(prop.node.key.id.name);
+  const { id, getId, setId, initAdded } = privateName;
   const value = prop.node.value || prop.scope.buildUndefinedNode();
+  if (!prop.isProperty() && (initAdded || !(getId || setId))) return;
```

Comment:
When would it not be a property?

Reply:
It wouldn't be a property if it's a static private accessor.

Reply:
Nit: could you extract `getId || setId` to an `isAccessor` boolean variable?
I find it hard to read all these nested boolean expressions.